### DISCUSSION TITLE
Stuck on last step of User Login OAuth Flow

### DISCUSSION THREAD

**Discussion by Koramchad** [gilesknap/gphotos-sync#385] [Q&A] [answered]
Trying to set this up on a Qnap following the instructions here: https://forum.qnap.com/viewtopic.php?t=159827&p=781241

Everything is working fine until the very last step, where I click both boxes to give the app access to my Google Photos account. Once I click continue, it tried to direct me back to my Qnap, but the link is broken:

<img width="718" alt="image" src="https://user-images.githubusercontent.com/1527414/190947229-2a7bbd64-284a-4001-af34-c946c79449d3.png">

I assume this is needed to connect to the app again to let it know that it has access to the google photos account, but I can't figure out how to link the last step.

Any ideas?

Thanks!!

**Comment by Koramchad** [comment]
I found this section on the tutorial:

> The authentication flow only allows localhost for security reasons so the first run must always be done on a machine with a browser.
> 
> If you are running on a NAS or other headless server you will first need to run locally so that you can do initial login flow with a browser. Then copy <TARGET>/.gphotos.token to the server. For this first run you could use the following options so that no backup is performed:
> 
> --skip-files --skip-albums --skip-index

What exactly is "copy TARGET/.gphotos.token to the server" and how do I do it? I've been copying the URL response I get from the login process and pasting it on a .gphotos.token file in the download and .cnofig folders, but so far nothing has triggered the terminal to continue the start up process.

Any help would be very much appreciated by my mental health.

Thanks!

**Comment by gilesknap** [comment, marked as answer]
OK, since the tutorial on Qnap was written, Google have deprecated the auth flow that gphotos-sync used to use.

What I'm recommending in the gphtotos-sync [tutorial](https://gilesknap.github.io/gphotos-sync/main/tutorials/installation.html#headless-gphotos-sync-servers) is that you do the first run locally on a workstation with a browser. TARGET is the target directory that you pass to gphotos-sync. So if you did 
```
gphotos-sync --skip-files --skip-albums --skip-index /home/me/myphotos
```
You would find the token file at `` /home/me/myphotos/.gphotos.token``

You then need to copy that to the target folder on your NAS.

An alternative approach is to follow the standard procedure on your NAS, paste the URL into a browser and then modify the URL when you get the site can't be reached message. See a description of this approach [here](https://github.com/gilesknap/gphotos-sync/issues/379#issuecomment-1221191662). This will only work if your NAS is contactable on port 80 from your browser and its not already using port 80.

By the way, I'm not sure why your error message is showing 192.168.7.70. I would expect it to show localhost or 127.0.0.1 not sure what is happening here unless the secrets file has been edited?

**Comment by Koramchad** [comment]
Perfect, this is exactly what i needed to move forward. I should have
figured out what you meant by trying this in a local machine first with
access to a browser only to then transplant the token to the NAS.
The error message with the IP instead of localhost was me trying what was
suggested on that approach you linked, replacing localhost with the NAS
address.


On Mon, Sep 19, 2022 at 12:13 PM Giles Knap ***@***.***>
wrote:

> OK, since the tutorial on Qnap was written, Google have deprecated the
> auth flow that gphotos-sync used to use.
>
> What I'm recommending in the tutorial is that you do the first run locally
> on a workstation with a browser. TARGET is the target directory that you
> pass to gphotos-sync. So if you did
>
> gphotos-sync --skip-files --skip-albums --skip-index /home/me/myphotos
>
> You would find the token file at /home/me/myphotos/.gphotos.token
>
> You then need to copy that to the target folder on your NAS.
>
> An alternative approach is to follow the standard procedure on your NAS,
> paste the URL into a browser and then modify the URL when you get the site
> can't be reached message. See a description of this approach here
> <https://github.com/gilesknap/gphotos-sync/issues/379#issuecomment-1221191662>.
> This will only work if your NAS is contactable on port 80 from your browser
> and its not already using port 80.
>
> By the way, I'm not sure why your error message is showing 192.168.7.70. I
> would expect it to show localhost or 127.0.0.1 not sure what is happening
> here unless the secrets file has been edited?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gilesknap/gphotos-sync/discussions/385#discussioncomment-3683969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALU45VSJILB4SNAKBKDAVDV7C3NDANCNFSM6AAAAAAQPXT25I>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 


Álvaro Gándara Astray

**Comment by drzraf** [comment]
I think a better solution would be to allow gphotos-sync to accept the flow's CODE value from copy/pasting. Many apps do it that way.

**Reply by gilesknap** [reply]
@drzraf I'm afraid that flow is no longer supported by google servers, although I'm not sure why.

**Comment by milan187** [comment]
I am having real issues with this.
I tried taking the token file thats still working on my synology to my qnap but the container is still coming up with URL to approve.
I tired copying all the files still the same, what can I be missing?

I also tried putting in the IP address of NA when I get error on Google Page, but thats giving me a proxy error. I am on the same network as the NAS. 

EDIT: Worked when I created a new token file on Linux and pasted to QNAP. Thanks for everything, its Google that sucks. But at least there is a way to do this. Thanks again!
